app/controllers/bookings_controller.rb

```diff
@@ -1,8 +1,49 @@
 class BookingsController < ApplicationController
+  before_action :find_booking, only: [:show :edit, :update, :destroy]
+
+  def index
+    @bookings = Booking.all
+  end
 
   def new
+    @booking = Booking.new
   end
 
   def create
+    @booking = Booking.new(booking_params)
+    @booking.user = @user
+    @booking.boat = @boat
+    if @booking.save
+      redirect_to user_bookings_path(@user)
+    else
+      render :new
+    end
+  end
+
+  def show
+  end
+
+  def edit
+  end
+
+  def update
+    @booking.update(booking_params)
+    redirect_to user_booking_path(@booking.user)
   end
+
+  def destroy
+    @booking.destroy
+    redirect_to user_booking_path(@booking.user)
+  end
+
+  private
+
+  def find_booking
+    @booking = Booking.find(params[:booking_id])
```

Comment:
I believe he booking is params[:id]. Please check the routes

Reply:
True, thank you.